### ISSUE TITLE
Simplify training: use Adam and remove the need for manual tuning

### ISSUE THREAD

**Issue by daniel-j-h** [mapbox/robosat#58]
At the moment we are using stochastic gradient descent and a multi-step weight decay policy.

In this setup the user has to set
- the initial sgd learning rate
- the sgd momentum to use
- the weight decay milestones
- the weight decay factor

https://github.com/mapbox/robosat/blob/2067cb7429ffd95efc3d9ace255b234831615d39/robosat/tools/train.py#L83-L85

And while this allows for great flexibility and control over details it might be to complicated for our users. We should look into replacing our current setup e.g. with the [Adam optimizer](https://pytorch.org/docs/stable/optim.html#torch.optim.Adam) only setting the initial learning rate and the weight decay.

We can then set these two values to reasonable defaults and users can get started without thinking too much about parameters and without having to run multiple experiments just to get basic parameters figured out.

Tasks
- [x] Implement Adam optimizer with learning rate and weight decay
- [x] Benchmark and check results; if it looks reasonable go for it
- [x] Remove sgd parameters from config; use learning rate and weight decay only
